### PR TITLE
feat: skip Android & Detox setup in ios_build job

### PR DESCRIPTION
**Pull request by devnev**

### CODE REVIEW

File: src/commands/setup_macos_executor.yml [hunk 1]
```diff
@@ -27,7 +35,12 @@ steps:
         source $BASH_ENV
 
   - when:
-      condition: <<parameters.homebrew_cache>>
+      condition:
+        and:
+          - <<parameters.homebrew_cache>>
+          - or:
+              - <<parameters.android>>
+              - <<parameters.detox>>
```
Comment (cortinico) [inline src/commands/setup_macos_executor.yml:43]: why are you adding this? If you expose the `homebrew_cache` to the user, that's the only key you should honor for the sake of the cache

File: src/commands/setup_macos_executor.yml [hunk 2]
```diff
@@ -50,24 +63,42 @@ steps:
...
+          command: |
+            HOMEBREW_NO_AUTO_UPDATE=1 brew tap wix/brew >/dev/null
+            HOMEBREW_NO_AUTO_UPDATE=1 brew install applesimutils >/dev/null
 
   - when:
-      condition: <<parameters.homebrew_cache>>
+      condition:
+        and:
+          - <<parameters.homebrew_cache>>
+          - or:
```
Comment (cortinico) [inline src/commands/setup_macos_executor.yml:99]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]

File: src/commands/setup_macos_executor.yml (hunk 1 above)
Comment (devnev) [reply src/commands/setup_macos_executor.yml:43]: If both android and detox are false, brew is never used, hence skipping both the update and the cache - seems silly to nonetheless do caching.

Does the same logic apply to homebrew_update - should brew update run even if detox and android are disabled?

Comment (cortinico) [reply src/commands/setup_macos_executor.yml:43]: I would just always keep brew update and allow the users to skip the `homebrew_cache` as they see fit. If you wish you can add another parameter for the `homebrew_update` step

Comment (devnev) [reply src/commands/setup_macos_executor.yml:43]: There already is a parameter for the homebrew update step.

In my own usage, once android and detox were gone, the brew update was adding extra time to the build for no benefit, so turning it off automatically in the orb seemed like the better approach.

Comment (cortinico) [reply src/commands/setup_macos_executor.yml:43]: Oh you're right. Still I won't turn automatically steps and will just give the user full control

**Comment by devnev** [comment]
ping @cortinico

**Pull request merged**